drivers/gpio/gpio_ifx_cat1.c

```diff
@@ -50,93 +48,58 @@ struct gpio_cat1_data {
 /* Get port number by calculation difference from current port address minus
  * GPIO base address divided by GPIO structure size.
  */
-#define GET_PORT_NUM(dev)				     \
-	(((uint32_t) ((const struct gpio_cat1_config *const) \
-		      (dev)->config)->regs - CY_GPIO_BASE) / GPIO_PRT_SECTION_SIZE)
-
-#define GET_DEV_OBJ_FROM_LIST(i, _) \
-	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio_prt##i))
-
-/* Map port number to device object */
-static const struct device *const port_dev_obj[IOSS_GPIO_GPIO_PORT_NR] = {
-	/* the integer used as the first variable in listify is equivalent to
-	 * IOSS_GPIO_GPIO_PORT_NR for the respective categories, but using
-	 * the macro in LISTIFY causes build failures
-	 */
-	#if CONFIG_SOC_FAMILY_INFINEON_CAT1A
-		LISTIFY(15, GET_DEV_OBJ_FROM_LIST, (,))
-	#elif CONFIG_SOC_FAMILY_INFINEON_CAT1B
-		LISTIFY(6, GET_DEV_OBJ_FROM_LIST, (,))
-	#endif
-};
+#define GET_PORT_NUM(dev)                                                                          \
```

Comment:
I would be interested in a comment that explains why this math works.

Reply:
let's remove GET_PORT_NUM, we should use  `base` instead Cy_GPIO_PortToAddr(GET_PORT_NUM(dev)). GET_PORT_NUM was needed for cyhal.